stacker_blueprints/eks.py

```diff
@@ -0,0 +1,427 @@
+
+    launch_config_userdata = [
+        "#!/bin/bash -xe\n",
+        "CA_CERTIFICATE_DIRECTORY=/etc/kubernetes/pki", "\n",
+        "CA_CERTIFICATE_FILE_PATH=$CA_CERTIFICATE_DIRECTORY/ca.crt", "\n",
+        "MODEL_DIRECTORY_PATH=~/.aws/eks", "\n",
+        "MODEL_FILE_PATH=$MODEL_DIRECTORY_PATH/eks-2017-11-01.normal.json", "\n",
+        "mkdir -p $CA_CERTIFICATE_DIRECTORY", "\n",
+        "mkdir -p $MODEL_DIRECTORY_PATH", "\n",
+        "curl -o $MODEL_FILE_PATH https://s3-us-west-2.amazonaws.com/amazon-eks/1.10.3/2018-06-05/eks-2017-11-01.normal.json",
+        "\n",
+        "aws configure add-model --service-model file://$MODEL_FILE_PATH --service-name eks", "\n",
+        "aws eks describe-cluster --region=", Ref("AWS::Region"), " --name=", cluster_name,
+        " --query 'cluster.{certificateAuthorityData: certificateAuthority.data, endpoint: endpoint}' > /tmp/describe_cluster_result.json",
+        "\n",
+        "cat /tmp/describe_cluster_result.json | grep certificateAuthorityData | awk '{print $2}' | sed 's/[,\"]//g' | base64 -d >  $CA_CERTIFICATE_FILE_PATH",
+        "\n",
+        "MASTER_ENDPOINT=$(cat /tmp/describe_cluster_result.json | grep endpoint | awk '{print $2}' | sed 's/[,\"]//g')",
+        "\n",
+        "INTERNAL_IP=$(curl -s http://169.254.169.254/latest/meta-data/local-ipv4)", "\n",
+        "sed -i s,MASTER_ENDPOINT,$MASTER_ENDPOINT,g /var/lib/kubelet/kubeconfig", "\n",
+        "sed -i s,CLUSTER_NAME,", cluster_name, ",g /var/lib/kubelet/kubeconfig", "\n",
+        "sed -i s,REGION,", Ref("AWS::Region"), ",g /etc/systemd/system/kubelet.service", "\n",
+        "sed -i s,MAX_PODS,", FindInMap("MaxPodsPerNode", instance_type, "MaxPods"),
+        ",g /etc/systemd/system/kubelet.service", "\n",
+        "sed -i s,MASTER_ENDPOINT,$MASTER_ENDPOINT,g /etc/systemd/system/kubelet.service", "\n",
+        "sed -i s,INTERNAL_IP,$INTERNAL_IP,g /etc/systemd/system/kubelet.service", "\n",
+        "DNS_CLUSTER_IP=10.100.0.10", "\n",
+        "if [[ $INTERNAL_IP == 10.* ]] ; then DNS_CLUSTER_IP=172.20.0.10; fi", "\n",
+        "sed -i s,DNS_CLUSTER_IP,$DNS_CLUSTER_IP,g  /etc/systemd/system/kubelet.service", "\n",
+        "sed -i s,CERTIFICATE_AUTHORITY_FILE,$CA_CERTIFICATE_FILE_PATH,g /var/lib/kubelet/kubeconfig", "\n",
+        "sed -i s,CLIENT_CA_FILE,$CA_CERTIFICATE_FILE_PATH,g  /etc/systemd/system/kubelet.service", "\n",
+        "systemctl daemon-reload", "\n",
+        "systemctl restart kubelet", "\n",
+        "/opt/aws/bin/cfn-signal -e $? ",
+        "         --stack ", Ref("AWS::StackName"),
+        "         --resource NodeGroup ",
+        "         --region ", Ref("AWS::Region"), "\n"
+    ]
+    return Base64(Join("", launch_config_userdata))
+
+
+class Workers(Blueprint):
+    VARIABLES = {
+        "ClusterName": {
+            "type": str,
+            "description": "The name of the cluster for workers to join."
+        },
+        "WorkerSecurityGroupId": {
```

Comment:
I'd drop the `Worker` part of these variables.  They're already attached to the `Worker` blueprint, and it makes it easier if you can specify generic terms like `Subnets` and `SecurityGroup` (which I believe are the variables we use elsewhere).

Reply:
done